connect-rest-extension/src/main/java/io/debezium/kcdrestextension/DebeziumResource.java

```diff
@@ -0,0 +1,137 @@
+    // but currently a worker simply leaving the group can take this long as well.
+    public static final long REQUEST_TIMEOUT_MS = 90 * 1000;
+    // Mutable for integration testing; otherwise, some tests would take at least REQUEST_TIMEOUT_MS
+    // to run
+    private static long requestTimeoutMs = REQUEST_TIMEOUT_MS;
+
+    private final List<TransformsInfo> transforms;
+    private Boolean isTopicCreationEnabled = null;
+    private final Herder herder;
+    private final Map<String, ?> config;
+
+    private static final ComparableVersion TOPIC_CREATION_KAFKA_VERSION = new ComparableVersion("2.6.0");
+
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public DebeziumResource(ConnectClusterState clusterState, Map<String, ?> config) {
+        Field herderField;
+        try {
+            herderField = ConnectClusterStateImpl.class.getDeclaredField("herder");
+        }
+        catch (NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+        herderField.setAccessible(true);
+        try {
+            this.herder = (Herder) herderField.get(clusterState);
+        }
+        catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        this.transforms = new ArrayList<>();
+        this.config = config;
+    }
+
+    // For testing purposes only
+    public static void setRequestTimeout(long requestTimeoutMs) {
+        DebeziumResource.requestTimeoutMs = requestTimeoutMs;
+    }
+
+    public static void resetRequestTimeout() {
+        DebeziumResource.requestTimeoutMs = REQUEST_TIMEOUT_MS;
+    }
+
+    @GET
+    @Path("/transforms")
+    public List<TransformsInfo> listTransforms() {
+        return this.getTransforms();
+    }
+
+    private synchronized List<TransformsInfo> getTransforms() {
+        if (this.transforms.isEmpty()) {
+            Iterator var1 = this.herder.plugins().transformations().iterator();
+
+            while (var1.hasNext()) {
+                PluginDesc<Transformation> plugin = (PluginDesc) var1.next();
+
+                if ("org.apache.kafka.connect.runtime.PredicatedTransformation".equals(plugin.className())) {
+                    this.transforms.add(new TransformsInfo(HasHeaderKey.class.getName(), (new HasHeaderKey<>().config())));
+                    this.transforms.add(new TransformsInfo(RecordIsTombstone.class.getName(), (new RecordIsTombstone<>().config())));
+                    this.transforms.add(new TransformsInfo(TopicNameMatches.class.getName(), (new TopicNameMatches<>().config())));
+                }
+                else {
+                    this.transforms.add(new TransformsInfo(plugin));
+                }
+            }
+        }
+
+        return Collections.unmodifiableList(this.transforms);
+    }
+
+    @GET
+    @Path("/topic-creation")
+    public boolean getTopicCreationEnabled() {
+        return isTopicCreationEnabled();
```

Comment:
Let's call it once in the constructor and make the field final.

Reply:
Please compare to https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResource.java

The reason for this might be there could be parallel threads initializing the (distributed) herder when booting up KCD. I just copied the original behavior from the ConnectorPluginsResource. Am I to skeptic here?

Reply:
Will do that.